s3migrate/lambda.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+import boto3
+import pymogilefs
+
+from shared import MogileFile, make_bucket_map
+
+
+def process(event, _context):
+    """Migrate this MogileFile to S3 in AWS Lambda."""
+    bucket_map = make_bucket_map(os.environ["BUCKETS"])
+    mogile_client = pymogilefs.client.Client(
+        trackers=os.environ['MOGILE_TRACKERS'].split(','),
+        domain='plos_repo')
+    s3_resource = boto3.resource('s3')
+    table = boto3.resource('dynamodb').Table(os.environ["DYNAMODB_TABLE"])
```

Comment:
If you move these assignments outside of the function, lambda will reuse them between function calls on a given container. It might help speed up lambda time, or reduce the connection init/teardown costs on mogile (which might be the only reason to do it, because IIRC lambda itself isn't a bottleneck here)

Reply:
507e5cb

Reply:
Just be warned that it might reuse them to the point that the the db and mogile connections get stale. Lambda's auto-retry will take care of trying again if that happens, but you might wind up with weird consequences if your changes aren't atomic

Reply:
Well, it's worth a try. The bottleneck seemed to be mogile in dev, but I don't know what the bottleneck will be in prod.